### DISCUSSION TITLE
How to search/filters mails/messages

### DISCUSSION THREAD

**Discussion by arnaudlemaignen** [microsoftgraph/msgraph-sdk-go#438] [Q&A]
Hi
I am trying to filter mails like in the example of graph Explorer.
My Issue is I am not able to find any references to the _graphconfig_ import
and _github.com/microsoftgraph/msgraph-sdk-go/me_ does not exist anymore

`requestSearch := "\"hello world\""

requestParameters := &graphconfig.MeMessagesRequestBuilderGetQueryParameters{
	Search: &requestSearch,
}
configuration := &graphconfig.MeMessagesRequestBuilderGetRequestConfiguration{
	QueryParameters: requestParameters,
}
result, err := graphClient.Me().Messages().Get(context.Background(), configuration)`

sdk version:
msgraph-sdk-go v0.56.0

I guess it is because the API has changed and doc is always the last thing to fix :(.
Can you point me to the new implementation?

thanks

**Comment by baywet** [comment]
Hi @arnaudlemaignen 
Thanks for reporting this.

I created an issue to go fix the snippets generation see #442

The updated code to fix your application should look something like

```go
requestSearch := ""hello world""

requestParameters := &graphconfig.UserItemMessagesRequestBuilderGetQueryParameters{
    Search: &requestSearch,
}
configuration := &graphconfig.UserItemMessagesRequestBuilderGetRequestConfiguration{
    QueryParameters: requestParameters,
}
result, err := graphClient.Me().Messages().Get(context.Background(), configuration)
```

**Reply by arnaudlemaignen** [reply]
Many thanks for the help
just one quick fix, the good syntax is 

graphconfig "github.com/microsoftgraph/msgraph-sdk-go/users"

requestSearch := "hello world"

requestParameters := &graphconfig.ItemMessagesRequestBuilderGetQueryParameters{
    Search: &requestSearch,
}
configuration := &graphconfig.ItemMessagesRequestBuilderGetRequestConfiguration{
    QueryParameters: requestParameters,
}


BTW my requestSearch has an underscore char which is not valid for the API
e.g. 
requestSearch := "hello_world_test"
when I execute I got this message
msg: Syntax error: character '_' is not valid at position 5 in 'hello_world_test'."

is there an escape code to use for char such as '_'?

tks

**Reply by baywet** [reply]
is that error message coming from the Go compiler or from the API itself?

**Reply by arnaudlemaignen** [reply]
It is the API

Here is the complete API error
error:error status code received from the API
code: BadRequest
msg: Syntax error: character '_' is not valid at position 5 in 'hello_world_test'."

**Reply by baywet** [reply]
We keep the questions/issues on this repository related to the SDK itself because the API is so vast that service teams are not watching all the client side repos. For questions about the service the best thing to do [is to post in the dedicated Microsoft Q&A forum](https://learn.microsoft.com/en-us/answers/tags/281/microsoft-graph-search-api) so the service team can answer you.

**Reply by arnaudlemaignen** [reply]
Actually it was my bad correct syntax is
requestSearch := "\\"hello_world_test\\""
